In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio
import plotly.colors as colors
pio.templates_default = "plotly_white"
from pathlib import Path

In [ ]:
my_path = Path('Superstore.csv')

if my_path.is_file() == False:
  !wget 'https://raw.githubusercontent.com/ChetanParmarHQ/E-commerce-Sales-Analysis-Python/refs/heads/main/Superstore.csv'
try:
  df = pd.read_csv('Superstore.csv', encoding = 'latin-1')
except Exception as e:
  print(e)

In [ ]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [ ]:
df.describe()

,Row ID,Postal Code,Sales,Quantity,Discount,Profit
count,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000
mean,4997.500000,55190.379428,229.858001,3.789574,0.156203,28.656896
std,2885.163629,32063.693350,623.245101,2.225110,0.206452,234.260108
min,1.000000,1040.000000,0.444000,1.000000,0.000000,-6599.978000
25%,2499.250000,23223.000000,17.280000,2.000000,0.000000,1.728750
50%,4997.500000,56430.500000,54.490000,3.000000,0.200000,8.666500
75%,7495.750000,90008.000000,209.940000,5.000000,0.200000,29.364000
max,9994.000000,99301.000000,22638.480000,14.000000,0.800000,8399.976000


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

#Converting date columns

In [ ]:
df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Ship Date'] = pd.to_datetime(df['Ship Date'])

In [ ]:
df[['Order Date', 'Ship Date']].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 2 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   Order Date  9994 non-null   datetime64[ns]
 1   Ship Date   9994 non-null   datetime64[ns]
dtypes: datetime64[ns](2)
memory usage: 156.3 KB


In [ ]:
df['Order Month'] = df['Order Date'].dt.month
df['Order Year'] = df['Order Date'].dt.year
df['Order Day of Week'] = df['Order Date'].dt.dayofweek

In [ ]:
df[['Order Date','Order Month', 'Order Year', 'Order Day of Week']].head()

,Order Date,Order Month,Order Year,Order Day of Week
0,2016-11-08,11,2016,1
1,2016-11-08,11,2016,1
2,2016-06-12,6,2016,6
3,2015-10-11,10,2015,6
4,2015-10-11,10,2015,6


# Monthly Sales Analysis

In [ ]:
sales_by_month = df.groupby('Order Month')['Sales'].sum().reset_index()

sales_by_month

,Order Month,Sales
0,1,94924.8356
1,2,59751.2514
2,3,205005.4888
3,4,137762.1286
4,5,155028.8117
5,6,152718.6793
6,7,147238.0970
7,8,159044.0630
8,9,307649.9457
9,10,200322.9847


In [ ]:
fig = px.line(sales_by_month,
              x = 'Order Month',
              y = 'Sales',
              title = 'Monthly Sales Analysis')
fig.show()

# Sales by Category

In [ ]:
sales_by_category = df.groupby('Category')['Sales'].sum().reset_index()

sales_by_category

,Category,Sales
0,Furniture,741999.7953
1,Office Supplies,719047.0320
2,Technology,836154.0330


In [ ]:
from IPython.utils import text
fig = px.pie(sales_by_category,
             values = 'Sales',
             names = 'Category',
             hole = 0.5)

fig.update_traces(textposition = 'inside', textinfo = 'percent+label')
fig.update_layout(title_text = 'Sales Analysis by Category', title_font=dict(size=24))

fig.show()

#Sales Analysis by Sub Category

In [ ]:
sales_by_subcategory = df.groupby('Sub-Category')['Sales'].sum().reset_index().sort_values(by = 'Sales', ascending = False)

sales_by_subcategory

,Sub-Category,Sales
13,Phones,330007.0540
5,Chairs,328449.1030
14,Storage,223843.6080
16,Tables,206965.5320
3,Binders,203412.7330
11,Machines,189238.6310
0,Accessories,167380.3180
6,Copiers,149528.0300
4,Bookcases,114879.9963
1,Appliances,107532.1610


In [ ]:
sales_by_subcategory

,Sub-Category,Sales
13,Phones,330007.0540
5,Chairs,328449.1030
14,Storage,223843.6080
16,Tables,206965.5320
3,Binders,203412.7330
11,Machines,189238.6310
0,Accessories,167380.3180
6,Copiers,149528.0300
4,Bookcases,114879.9963
1,Appliances,107532.1610


In [ ]:
fig = px.bar(sales_by_subcategory,
             x = 'Sub-Category',
             y = 'Sales',
             title = 'Sales Analysis by Sub-Category')
fig.show()

# Monthly Profit Analysis

In [ ]:
profit_by_month = df.groupby('Order Month')['Profit'].sum().reset_index()

profit_by_month

,Order Month,Profit
0,1,9134.4461
1,2,10294.6107
2,3,28594.6872
3,4,11587.4363
4,5,22411.3078
5,6,21285.7954
6,7,13832.6648
7,8,21776.9384
8,9,36857.4753
9,10,31784.0413


In [ ]:
fig = px.bar(profit_by_month,
             x = 'Order Month',
             y = 'Profit',
             title = 'Monthly Profit Analysis')
fig.show()

# Profit by Category

In [ ]:
profit_by_category = df.groupby('Category')['Profit'].sum().reset_index()

profit_by_category

,Category,Profit
0,Furniture,18451.2728
1,Office Supplies,122490.8008
2,Technology,145454.9481


In [ ]:
fig = px.pie(profit_by_category,
             values = 'Profit',
             names = 'Category',
             hole = 0.5)

fig.update_traces(textposition = 'inside', textinfo = 'percent+label')
fig.update_layout(title_text = 'Profit by Category', title_font = dict(size=24))

fig.show()

#Profit by Sub Category

In [ ]:
profit_by_subcategory = df.groupby('Sub-Category')['Profit'].sum().reset_index().sort_values(by = 'Profit', ascending = False)

profit_by_subcategory

,Sub-Category,Profit
6,Copiers,55617.8249
13,Phones,44515.7306
0,Accessories,41936.6357
12,Paper,34053.5693
3,Binders,30221.7633
5,Chairs,26590.1663
14,Storage,21278.8264
1,Appliances,18138.0054
9,Furnishings,13059.1436
7,Envelopes,6964.1767


In [ ]:
fig = px.bar(profit_by_subcategory,
             x = 'Sub-Category',
             y = 'Profit',
             title = 'Profit Analysis by Sub Category')
fig.show()

# Sales and Profit - Customer Segment

In [ ]:
df['Segment'].value_counts()

,count
Segment,
Consumer,5191
Corporate,3020
Home Office,1783


In [ ]:
sales_profit_by_segment = df.groupby('Segment').agg({'Sales':'sum', 'Profit':'sum'}).reset_index()

sales_profit_by_segment

,Segment,Sales,Profit
0,Consumer,1.161401e+06,134119.2092
1,Corporate,7.061464e+05,91979.1340
2,Home Office,4.296531e+05,60298.6785


In [ ]:
fig = go.Figure()

fig.add_trace(go.Bar(x = sales_profit_by_segment['Segment'],
                     y = sales_profit_by_segment['Sales'],
                     name = 'Sales'))
fig.add_trace(go.Bar(x = sales_profit_by_segment['Segment'],
                     y = sales_profit_by_segment['Profit'],
                     name = 'Profit'))
fig.update_layout(title = 'Sales and Profit Analysis by Segment',
                  xaxis_title = 'Custoemr Segment', yaxis_title = 'Amount')
fig.show()

# Sales to Profit Ratio

In [ ]:
# For Segment

sales_profit_by_segment['sales_profit_ratio'] = sales_profit_by_segment['Sales']/sales_profit_by_segment['Profit']

print(sales_profit_by_segment[['Segment', 'sales_profit_ratio']])

       Segment  sales_profit_ratio
0     Consumer            8.659471
1    Corporate            7.677245
2  Home Office            7.125416


In [ ]:
# For Category

sales_profit_by_category = df.groupby('Category').agg({'Sales':'sum', 'Profit':'sum'}).reset_index()

sales_profit_by_category['sales_profit_ratio'] = sales_profit_by_category['Sales']/sales_profit_by_category['Profit']

print(sales_profit_by_category[['Category', 'sales_profit_ratio']])

          Category  sales_profit_ratio
0        Furniture           40.214017
1  Office Supplies            5.870213
2       Technology            5.748543


In [ ]:
# For Sub Category

sales_profit_by_subcategory = df.groupby('Sub-Category').agg({'Sales':'sum', 'Profit': 'sum'}).reset_index()

sales_profit_by_subcategory['sales_profit_ratio'] = sales_profit_by_subcategory['Sales']/sales_profit_by_subcategory['Profit']

print(sales_profit_by_subcategory[['Sub-Category', 'sales_profit_ratio']])

   Sub-Category  sales_profit_ratio
0   Accessories            3.991267
1    Appliances            5.928555
2           Art            4.154362
3       Binders            6.730671
4     Bookcases          -33.082259
5        Chairs           12.352277
6       Copiers            2.688491
7     Envelopes            2.365879
8     Fasteners            3.185068
9   Furnishings            7.022295
10       Labels            2.251305
11     Machines           55.909076
12        Paper            2.304581
13       Phones            7.413268
14      Storage           10.519547
15     Supplies          -39.251163
16       Tables          -11.676159


#**Conclusion**

**Q1. You need to calculate the monthly sales of the store and identify which month had the highest sales and which month had the lowest sales.**
- Highest Sales Month: November (Month 11)
Sales: 352,461.07
- Lowest Sales Month: February (Month 2)
Sales: 59,751.25

**Q2. You need to analyze sales based on product categories and determine which category has the lowest sales and which category has the highest sales.**
- Highest Sales Category: Technology
Sales: 836,154.03 (36.4%)
- Lowest Sales Category: Office Supplies
Sales: 719,047.03 (31.3%)

**Q3. The sales analysis needs to be done based on sub-categories**
- Highest Selling Sub-Category: Phones
Sales: 330,007.05
- Lowest Selling Sub-Category: Fasteners
Sales: 3,024.28

**Q4. You need to analyze the monthly profit from sales and determine which month had the highest profit.**
- Highest Profit Month: December (Month 12)
Profit: 43,369.19
- Lowest Profit Month: January (Month 1)
Profit: 9,134.45

**Q5. Analyze the profit by category and sub-category.**
- By Category
  - Highest Profit Category: Technology
Profit: 145,454.95 (50.8%)
  - Lowest Profit Category: Furniture
Profit: 18,451.27 (6.44%)

- By Sub-Category
  - Highest Profit Sub-Category: Copiers
Profit: 55,617.82
  - Lowest Profit Sub-Category: Tables
Profit: -17,725.48 (Loss)

**Q6. Analyze the sales and profit by customer segment**
- Top Segment (Sales & Profit): Consumer
- Lowest Segment (Sales & Profit): Home Office

**Q7. Analyze the sales to profit ratio**
- By Segment
  - Consumer: 8.66
  - Corporate: 7.68
  - Home Office: 7.13 ✅ (Best profitability)
- By Category
  - Furniture: 40.21 ❌ (Very low profitability)
  - Office Supplies: 5.87
  - Technology: 5.75 ✅ (Best profitability)
- By Sub-Category
  - Best Profitability (Low Ratio):
Envelopes: 2.37,
Labels: 2.25,
Paper: 2.30
  - Worst Profitability (High/Negative Ratio):
Machines: 55.91,
Bookcases: -33.08,
Supplies: -39.25,
Tables: -11.68